lib/lwm2m-server.js

```diff
@@ -103,6 +104,29 @@ function validateTypes(config, callback) {
     callback(error);
 }
 
+/**
+ * If Lifetime Resource exists, then the registration SHOULD be removed by the Server if a new registration or update
+ * is not received within this lifetime.
+ */
+function checkLifetime() {
+    var CHECK_INTERVAL = 1000;
+
+    return setInterval(function(){
+        registry.list(function(error, deviceList){
+            if (!error && deviceList) {
+                deviceList.forEach(function(device){
+                    if (device.lifetime &&
+                        new Date() - new Date(device.creationDate) > Number(device.lifetime) * 1000) {
+                        registry.unregister(device.id, function(err, obj){
+
```

Comment:
A debug log would be appropriate at this point, for the cases where the unregister operation succeeds. It should check the error parameter and show an error log in case it was not able to unregister the device.